### ISSUE TITLE
Build ApiControllers to return data the app can consume

### ISSUE THREAD

**Issue by kdinev** [dgiormov/ista.agenda.builder.web#40]
The current app utilizes a back-end, which we need to switch from and make our current website deliver the necessary data. Build API controllers to handle agenda building and feedback collection with our current back-end

**Comment by kdinev** [comment]
Depends on #39

**Comment by AndreyKarabeljov** [comment]
The ApiControllers are available inside the Rest/Controllers - the view models are in Rest/ViewModels

**Comment by AndreyKarabeljov** [comment]
I started with the sessions api

**Comment by AndreyKarabeljov** [comment]
Some of the basic logic around sessions is implemented. There is a session controller which retrieves the sessions, session details and the participant sessions from the database. 

Note that there are some properties missing from the MVC speaker`s schema like – company, city and job_title. I guess for now we could remove the logic related to them.

Additionally there are some fields which appear to be not used anywhere, e.g. – speaker.clickedUpon, session.hasSecret, session.tags, session.searchTerms, session.date, etc.

**Comment by AndreyKarabeljov** [comment]
I am starting with the Session/Speaker rate

**Comment by dgiormov** [comment]
@AndreyKarabeljov many of these fields are not used and will be ignored if missing.
I will doublecheck and remove the functionality that is affected.

Update:
Just checked the UI works without those fields.

**Comment by AndreyKarabeljov** [comment]
@dgiormov can you please clarify what is the purpose of WaitForIt and Available  Feedback properties and whether they are need?

**Comment by dgiormov** [comment]
@AndreyKarabeljov waitFoIt is when you want to postpone the time when the feedback can be given, for example we had check that did not allow providing feedback before the start of the conference (so appropriate message will be shown to the user. Something like: You can post your feedback after the conference has started)

Available is a marker to hide the questions, either after providing feedback or when the feedback timeslot is still not opened. So for example when you post your feedback available will become false, the questions will be hidden and message like "Thank you for your feedback" will surface. 
If we want to provide the possibility that the user can post feedback multiple times the available should return always true, if we want to limit her to only one feedback it should return false once the feedback is recorded.

**Comment by AndreyKarabeljov** [comment]
Thanks, I will add an application setting to control both properties.
Regarding the multiple feedbacks, I implemented to be possible to add new feedback which overrides the previous.

**Comment by AndreyKarabeljov** [comment]
The app is deployed in production. Here are the Angular app remaining issues:
- ~~Partners screen images are not styled very well and their sizes are defined by inline css~~
- ~~There are some javascript errors which should be fixed~~
- ~~Partners screen update~~
- ~~Some speakers`s pics are too huge~~
- ~~The logout should be done~~
- ~~Home page is the Welcome screen while it should be the Agenda screen~~
- ~~The track grouping by time should be fixed on the Agenda screen~~
- ~~There are some missing resources when requesting pages.~~
- ~~The home icon should redirect to the MVC app home page~~

Some remaining activities:
- ~~A preliminary testing environment should be done using prod data~~
- ~~Check whether the original main.scss file can be used. Investigate how VS could handle SASS files.~~

**Comment by kdinev** [comment]
I implemented the logout in the view.

**Comment by kamen-stankov** [comment]
Some speakers`s pics are too huge fixed + speaker-info section removed (we don't have such info)

**Comment by kamen-stankov** [comment]
fixed - Home page is the Welcome screen while it should be the Agenda screen

**Comment by kamen-stankov** [comment]
fixed - The home icon should redirect to the MVC app home page

**Comment by kamen-stankov** [comment]
Partners screen update - Partners api controller created. I wanted to copy the angular controller from here, but then I saw that partners template is static page and there is no controller. So could someone, more familiar with angular, create a controller and Andrey will adopt the template.

**Comment by AndreyKarabeljov** [comment]
The partners html template is implemented however the images are not displayed very well and some inline styling is present so it should be fixed

**Comment by kdinev** [comment]
I think I've fixed it now.

**Comment by dgiormov** [comment]
Should we remove the "Hello, " in the menu?
As I saw it is possible to rate right now, shall we have some time-stoppers?

**Comment by kdinev** [comment]
The Hello is OK, but we will have to restrict the feedback from the morning of the conference day until a week later.

**Comment by AndreyKarabeljov** [comment]
The feedback is restricted by a param in the config file, but it might be better to restrict it on the fly. 
I think @dgiormov asked for the rating functionality which currently doesn`t have any restrictions. Is there any mechanism to do that in the current Angular implementation?

Additionally, where should be added the mobile app link to the MVC app? Should we replace the current Agenda link or introduce new one?

I also fixed some bugs so they could be deployed on prod.

**Comment by AndreyKarabeljov** [comment]
I updated both feedback and ratings to calculate if they are active based on the current date. They will be active from 18-11 8:00 to 27-11 00:00. @kdinev Can you please make sure that the server time is in the right time zone?

**Comment by dgiormov** [comment]
@kdinev  & @AndreyKarabeljov should we remove the comments?

**Comment by AndreyKarabeljov** [comment]
@dgiormov Why do we need to remove them?

**Comment by dgiormov** [comment]
@AndreyKarabeljov - I was not sure if there will be support for them in the new backend. If they work it is fine:)

**Comment by AndreyKarabeljov** [comment]
:) I thought you`ve found some issues with the functionality that stops its usage. Yep, there is a backend implementation for the comments

**Comment by dgiormov** [comment]
One more thing:)
Share buttons (twitter) points to #Ista2014 
and the session name is missing in the tweet

**Comment by kdinev** [comment]
I will change the hash tag. The article URL and bareArticleUrl entries are empty from the API I suppose?

**Comment by kdinev** [comment]
The URL seems good actually. It just contains a # after agenda. I guess we should encode that hashtag.

**Comment by dgiormov** [comment]
maybe some symbol is misinterpreted and the url is split from the browser
